apps/api/v2/src/ee/event-types/event-types.repository.ts

```diff
@@ -0,0 +1,34 @@
+import { MembershipsRepository } from "@/modules/memberships/memberships.repository";
+import { PrismaReadService } from "@/modules/prisma/prisma-read.service";
+import { Injectable, NotFoundException } from "@nestjs/common";
+import { User } from "@prisma/client";
+
+import { getEventTypeById } from "@calcom/platform-libraries";
+
+@Injectable()
+export class EventTypesRepository {
+  constructor(
+    private readonly dbRead: PrismaReadService,
+    private readonly membershipsRepository: MembershipsRepository
+  ) {}
+
+  async getUserEventType(user: User, eventTypeId: number) {
+    try {
+      const isUserOrganizationAdmin = user?.organizationId
+        ? await this.membershipsRepository.isUserOrganizationAdmin(user.id, user.organizationId)
+        : false;
+
+      const eventType = await getEventTypeById({
+        eventTypeId,
+        userId: user.id,
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
```

Comment:
I tried refactoring "getEventTypeById" for it to accept prisma client without extensions (the PrismaClient it now accepts is one from @calcom/prisma") but extended clients do not have $on and $use methods, and extended prisma has accelerate which changes the type even more.

since the prisma operations in getEventTypebyId are simple i think its fine to ignore the type problem here

Reply:
`
import type {PrismaClient} from '@calcom/prisma'


this.dbRead.prisma as unknown as PrismaCLient`